proposals/ssss-distributed-actor-runtime.md

```diff
@@ -0,0 +1,1798 @@
+
+```swift
+distributed actor DA {
+  // let id: ID
+  // let actorSystem: ActorSystem
+  
+  init(system: Self.ActorSystem) {
+    // ~~~ synthesized ~~~
+    self.actorSystem = system
+    // ... 
+    // ~~~ end of synthesized ~~~
+  }
+}
+```
+
+The initialization of the `id` property is a little more involved. We need to communicate with the `system` used to initialize the distributed actor, for it is the `ActorSystem` that allocates and manages identifiers. In order to obtain a fresh `ID` for the actor being initialized, we need to call `system`'s `assignID` method. This is done before any user-defined code is allowed to run in the actors designated initializer, like this:
+
+```swift
+distributed actor DA { 
+  let number: Int
+  // user defined:
+  init(system: ActorSystem) {
+    self.number = 42
+  }
+  
+  // ~~~ synthesized ~~~
+  init(system: ActorSystem) {
+    // ~~ injected property initialization ~~
+    self.actorSystem = system
+    self.id = system.assignID(Self.self)
+    // ~~ end of injected property initialization ~~
+
+    // user-defined code follows...
+    self.number = 42
+    
+    // ...
+  }
+  // ~~~ end of synthesized ~~~
+}
+```
+
+### Ready-ing Distributed Actors
+
+So far, the initialization process was fairly straight forward. We only needed to find a way to initialize the stored properties, and that's it. There is one more step though that is necessary to make distributed actors work: "ready-ing" the actor.
+
+As the actor becomes fully initialized, the type system allows escaping its `self` through method calls or closures. This can lead to unsafe access patterns, but those are discussed and prevented by [SE-0327: On Actor Initialization](https://github.com/apple/swift-evolution/blob/main/proposals/0327-actor-initializers.md). Distributed actor initializers are subject to the same rules as outlined in that proposal. In addition to that, in order for distributed actors to be able to escape through distributed calls made during their initializer, a system must carefully manage IDs that have been assigned but are not "ready" yet.
+
+Readying is done automatically by code synthesized into the distributed actor's non-delegating initializers, and takes the shape of an `self.actorSystem.actorReady(self)` call, injected at appropriate spots in the initializer. This call is necessary in order for the distributed actor system to be able to resolve an incoming `ID` (that it knows, since it assigned it) to a specific distributed actor instance (which it does not know, until `actorReady` is called on the system). This means that there is a state between the `assignID` and `actorReady` calls, during which the actor system cannot yet properly resolve the actor. 
+
+In this section we will discuss the exact semantics of readying actors, how systems must implement and handle these `actorReady` calls, and where exactly they are invoked during from any distributed actor initializer.
```

Comment:
We can simplify this whole section now to say that it will be called before `self` can escape the initializer. Some things that count as escapes include making a copy of `self`, or being passed as an argument, etc. You can refer people to the same situations that cause `self` to lose access to its stored properties from SE-327. In the event of no such uses of self, then the call happens before the return from the init.

Reply:
Awesome, great simplification and I cut a lot of content here thanks to this, thanks!